azurelinuxagent/common/exception.py

```diff
@@ -80,8 +80,9 @@ class AgentUpdateError(AgentError):
     When agent failed to update.
     """
 
-    def __init__(self, msg=None, inner=None):
+    def __init__(self, msg=None, inner=None, log_error=True):
         super(AgentUpdateError, self).__init__(msg, inner)
+        self.log_error = log_error
```

Comment:
this log_error property is really weird:

AgentUpdateHandler.run has the 'ext_gs_updated' flag, which it passes to _get_agent_family_manifest, which passes it to AgentUpdateError with a different name, 'log_error', which is handled by AgentUpdateHandler.

Seems like AgentUpdateHandler could just use ext_gs_updated directlry